app/services/concerns/xccdf/rule_group_rules.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+module Xccdf
+  # Methods related to saving RuleGroupRules
+  module RuleGroupRules
+    extend ActiveSupport::Concern
+
+    included do
+      def save_rule_parents
+        ::RuleGroupRule.import!(rules_with_rule_group_parent.select(&:new_record?), ignore: true)
+
+        rule_parents_base = ::RuleGroupRule.joins(rule: :benchmark).where(benchmark: { ref_id: @benchmark&.ref_id })
+        rule_parent_links_to_remove(rule_parents_base).delete_all
+      end
+
+      def save_rule_group_rules
+        save_rule_parents
+        save_rule_group_relationships
+      end
```

Comment:
The reason I had to split a couple of the methods into a separate method is because the method `save_all_benchmark_info` in util.rb was too many lines so I'm not sure how to get around that rubocop error besides doing something like this

Reply:
My suggestion for this is to wrap the big method using:
```
# rubocop:disable Metrics/MethodLength

# rubocop:enable Metrics/MethodLength
```

This is unorthodox and in any other case I'd be against it, but here we do a list of operations and I'd like to have it flat without side effects. If anyone else will complain, we can convert it into a dispatch array, but for now just suspend the rule please.

Reply:
I still think this should be flattened as I described above ^